src/main/java/uk/gov/hmcts/cmc/claimstore/utils/CaseDetailsConverter.java

```diff
@@ -41,4 +66,14 @@ private CCDCase extractCCDCase(Map<String, Object> mapData) {
     public Map<String, Object> convertToMap(CCDCase ccdCase) {
         return (Map<String, Object>) jsonMapper.convertValue(ccdCase, Map.class);
     }
+
+    private LocalDate calculateIntentionToProceedDeadline(LocalDateTime respondedDate) {
+        LocalDate deadline = respondedDate.toLocalDate().plusDays(this.intentionToProceedDeadline);
+
+        while (!workingDayIndicator.isWorkingDay(deadline)) {
```

Comment:
FYI - To get next working day we have this method to be used `workingDayIndicator.getNextWorkingDay(...)` which has unit tests as well.